.github/workflows/build.yaml

```diff
@@ -0,0 +1,166 @@
+            puppet: "6.0"
+            check: "parallel_spec"
+            name: 'spec'
+          - ruby: '2.5'
+            puppet: "6.0"
+            check: 'strings:generate[,,,,,,"--fail-on-warning"]'
+            name: "documentation"
+    env:
+      CHECK: '${{ matrix.check }}'
+      PUPPET_GEM_VERSION: '~> ${{ matrix.puppet }}'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '${{ matrix.ruby }}'
+      - name: Bundle prep
+        run: |
+          gem install bundler
+          bundle -v
+          rm -f Gemfile.lock
+          # Update system gems if requested. This is useful to temporarily workaround troubles in the test runner"
+          # See https://github.com/puppetlabs/pdk-templates/commit/705154d5c437796b821691b707156e1b056d244f for an example of how this was used"
+          # Ignore exit code of SIGPIPE'd yes to not fail with shell's pipefail set"
+          [ -z "$RUBYGEMS_VERSION" ] || (yes || true) | gem update --system $RUBYGEMS_VERSION
+          gem --version
+          bundle -v
+          bundle config path vendor/bundle
+          bundle config without 'system_tests'
+          bundle lock
+        # restore cache AFTER doing 'bundle lock' so that Gemfile.lock exists
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-${{ matrix.puppet }}-${{ matrix.ruby }}-gems-test-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.puppet }}-${{ matrix.ruby }}-gems-test-
+      - name: Bundle install
+        run: |
+          bundle install --jobs $(nproc) --retry 3
+      - name: Test
+        run: 'bundle exec rake $CHECK'
+
+  unit-python:
+    name: '${{matrix.name}} - python (${{matrix.python}})'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # note: actions/setup-python only allows using a major.minor releases
+          - python: '2.7'
+            make_target: 'python2'
+            name: 'bolt tasks'
+          - python: '3.6'
+            make_target: 'python3'
+            name: 'bolt tasks'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '${{ matrix.python }}'
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ matrix.puppet }}-{{ matrix.python }}-pip-unit-${{ hashFiles('**/requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.puppet }}-${{ matrix.ruby }}-pip-unit-
+      - name: Python prep
+        run: |
+          pip install virtualenv
+      - name: Test
+        run: 'make ${{ matrix.make_target }}'
+
+
+# TODO create a job for doing integration tests
```

Comment:
Tried to find this in CI checks, but figured the WIP.
That would be the most interesting part to compare and see how it works! :+1: